r/tests/testthat/helper-filesystems.R

```diff
@@ -0,0 +1,177 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+test_filesystem <- function(name, fs, path_formatter, uri_formatter) {
+  test_that(sprintf("read/write Feather on %s", name), {
+    write_feather(example_data, uri_formatter("test.feather"))
+    expect_identical(read_feather(uri_formatter("test.feather")), example_data)
+  })
+
+  test_that("read/write Feather by filesystem, not URI", {
```

Comment:
TODO: we need to name the filesystem in each test so we can differentiate test runs in CI output.